fvm/src/call_manager/default.rs

```diff
@@ -294,14 +330,15 @@ where
             };
 
             // From this point on, there are no more syscall errors, only aborts.
-            let result: std::result::Result<RawBytes, Abort> = (|| {
+            let mut result: std::result::Result<RawBytes, Abort> = (|| {
                 // Lookup the invoke method.
                 let invoke: wasmtime::TypedFunc<(u32,), u32> = instance
                     .get_typed_func(&mut store, "invoke")
                     // All actors will have an invoke method.
                     .map_err(Abort::Fatal)?;
 
                 // Invoke it.
+                // TODO: I'm not sure how to actually check the Trap here for OutOfFuel (as opposed to any other failure).
```

Comment:
This is the biggest open question.

Reply:
This is the underlying method in Wasmtime, but hooowww can I sniff this out 

```
    fn consume_fuel(&mut self, fuel: u64) -> Result<u64> {
        let consumed_ptr = unsafe { &mut *self.interrupts.fuel_consumed.get() };
        match i64::try_from(fuel)
            .ok()
            .and_then(|fuel| consumed_ptr.checked_add(fuel))
        {
            Some(consumed) if consumed < 0 => {
                *consumed_ptr = consumed;
                Ok(u64::try_from(-consumed).unwrap())
            }
            _ => bail!("not enough fuel remaining in store"),
        }
    }
```

Reply:
This is where fuel is consumed when _you_ manually call `consume_fuel`. When the actor runs out of fuel, I believe you get:

https://github.com/bytecodealliance/wasmtime/blob/6c8c94723a842258a4ead8e92eeb0a745a57aa64/crates/wasmtime/src/store.rs#L1805

But looking a the code... it looks like wasmtime will:

1. Consume the amount of fuel necessary to perform some operation.
2. Check to make sure it didn't go over the budget.
3. If it did, it'll abort.

So I _think_ we can just check `fuel_consumed > gas_to_fuel(gas_remaining)`. Specifically, we need to check that when the wasm call _fails_.